go/billing/api.py

```diff
@@ -247,6 +250,32 @@ def create_transaction_interaction(self, cursor, account_number,
         session_len_credits = MessageCost.calculate_session_length_credit_cost(
             session_len_cost, markup_percent)
 
+        query = """SELECT credit_balance, last_topup_balance
+                   FROM billing_account
+                   WHERE account_number = %(account_number)s"""
+
+        params = {'account_number': account_number}
+        cursor = yield cursor.execute(query, params)
+        result = yield cursor.fetchone()
+
+        if result is None:
+            raise BillingError(
+                "Unable to find billing account %s while checking"
+                " credit balance. Message was %s to/from tag pool %s." % (
+                    account_number, message_direction, tag_pool_name))
```

Comment:
This was moved up from below, so there is already a test case for this, buried in the huge `test_transaction` test: https://github.com/praekelt/vumi-go/blob/feature/issue-1210-billing-api-to-block-accounts-with-insufficient-credits/go/billing/tests/test_api.py#L536